crates/bevy_animation/src/lib.rs

```diff
@@ -123,23 +123,50 @@ impl AnimationClip {
     }
 }
 
+/// Repetition behavior of an animation.
+#[derive(Reflect, Copy, Clone)]
+pub enum RepeatAnimation {
+    /// The animation will never finish.
+    Forever,
+    /// The animation will finish after running once.
+    Never,
+    /// The animation will finish after running "n" times.
+    Count(u32),
+}
+
 #[derive(Reflect)]
 struct PlayingAnimation {
-    repeat: bool,
+    repeat: RepeatAnimation,
     speed: f32,
     elapsed: f32,
-    animation_clip: Handle<AnimationClip>,
+    animation_clip: Option<Handle<AnimationClip>>,
     path_cache: Vec<Vec<Option<Entity>>>,
+    /// Number of times the animation has completed.
+    /// If the animation is playing in reverse, this increments when the animation passes the start.
+    completions: u32,
 }
 
 impl Default for PlayingAnimation {
     fn default() -> Self {
         Self {
-            repeat: false,
+            repeat: RepeatAnimation::Never,
             speed: 1.0,
             elapsed: 0.0,
             animation_clip: Default::default(),
             path_cache: Vec::new(),
+            completions: 0,
+        }
+    }
+}
+
+impl PlayingAnimation {
+    /// Predicate to check if the animation has finished, based on its repetition behavior and the number of times it has repeated.
+    /// Note: An animation with `RepeatAnimation::Forever` will never finish.
+    pub fn finished(&self) -> bool {
```

Comment:
this could probably benefit from an inline hint

Reply:
Unsure as to what you mean by this? `#[inline]`?